### PR TITLE
Switch to eslint

### PR DESCRIPTION
**Pull request by xPaw**
Mostly `standard` with some rules to match what's currently used in the code. This makes line endings and ident consistent, so I recommend viewing the diff with whitespace changes hidden.

Two issues I didn't fix:
1. `ping_timer` in `startPeriodicPing` is assigned but never used, is it supposed to be cancelled somewhere?
2. This if check doesn't use triple equals and has jshint ignore, any reason why?

https://github.com/kiwiirc/irc-framework/blob/43e313dec062a2b5f7a19f93a4738001313337e0/src/transports/net.js#L234

I backtracked it to https://github.com/prawnsalad/KiwiIRC/commit/27bca914b2b921d6b0b08e2d6afac0cc9c9cfe80 but I'm not sure what the regression is?

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -203,16 +198,16 @@ module.exports = class IrcClient extends EventEmitter {
             // Add a reply() function to selected message events
             if (['privmsg', 'notice', 'action'].indexOf(event_name) > -1) {
                 event_arg.reply = function(message) {
-                    var dest = event_arg.target === client.user.nick ?
-                        event_arg.nick :
-                        event_arg.target;
+                    var dest = event_arg.target === client.user.nick
```
Comment (prawnsalad) [inline src/client.js]: Ternary ops should use the existing format in other Kiwi projects:
~~~javascript
foo = somevar === true ?
    'truthy' :
    'falsy';
~~~

**Comment by xPaw** [comment]
I wanted to leave let/const for a future PR so it was easier to keep track of the changes.

Can you look at the two questions?

File: src/client.js (hunk 1 above)
Comment (xPaw) [reply src/client.js]: Changed.

https://github.com/kiwiirc/irc-framework/compare/d0d223c567c19334cca0cdc71eb7f6966fe110a2..6b8671a311e9e183efe2caa219ad01f71b0b41f4

**Comment by prawnsalad** [comment]
> ping_timer in startPeriodicPing is assigned but never used, is it supposed to be cancelled somewhere?

This was changed in https://github.com/kiwiirc/irc-framework/commit/e5c3598dcb16b3839aba82b09b311aecae6b9408 and it's no longer needed to be stored anywhere. So `ping_timer` can safely be removed.

> This if check doesn't use triple equals and has jshint ignore, any reason why?

I remember getting confused with this line when it was originally put in way back then. Can't remember why now... @M2Ys4U if you see this, can you remember? Also, hi

**Comment by xPaw** [comment]
> So ping_timer can safely be removed.

Perhaps `that.connection.setTimeout` should be used there?

**Comment by xPaw** [comment]
I figured it out, `iconv.encode` returns a `Buffer`, not a string.

**Comment by xPaw** [comment]
I just tested all encodings in iconv-lite and there isn't a single one that fails the `TEST` ascii check. Can we just remove it?

Gotta keep `iconv.encode` in a try/catch though, so it still validates that the encoding actually exists.

**Pull request merged**